### ISSUE TITLE
FragmentFilter passTexture is always a RectangleTexture

### ISSUE THREAD

**Issue by devon-o** [Gamua/Starling-Framework#778]
In the FragmentFilter.updatePassTextures() method, the pass textures are created with the call: Texture.empty(width, height, PMA, false, true, scale). This means that unless using the 'baselineConstrained' profile, a RectangleTexture is always created. While this is nice performance-wise, a RectangleTexture cannot be wrapped/repeated - so if the AGAL shader in the filter being applied uses the 'repeat' or 'wrap' flag in the texture sample, the filter will fail. I haven't found a clean solution to suggest, but it seems the pass textures should be created with the repeat property set to the repeat property of the filtered display object's texture.

**Comment by neolit123** [comment]
hello Devon,

i think you have a point about the optional repeat, but a FragmentFilter target DisplayObject can also be a Quad, which doesn't have a Texture, so it can't really be obtained reliably from any target DisplayObject. also the target can be a deep tree of containers and a mixture of Texture DisplayObjects which have different texture settings.

it might be a good idea to simply expose the "repeat" (and/or "mipMapping") argument(s) which is (are) passed to Texture.empty() in FragmentFilter.updatePassTextures() as getter/setters in FragmentFilter. setting at least the optional "repeat" to "true" would not create a RectangleTexture for Context3D in Texture.empty().

if you modify FragmentFilter in such a way, does it work for your test case?
not sure how that would resonate as a whole...

**Comment by devon-o** [comment]
Good point about the quads and nested DO's. Adding a protected boolean 'repeat' property to the FragmentFilter (in a similar way as 'mode', 'numPasses', etc) does the trick. You can just set 'repeat' to true in your own FragmentFilter implementation and everything is fine. I may do a pull request later after I take another look with fresher eyes.

d.
